.devcontainer/install-dependencies.sh

```diff
@@ -308,7 +314,12 @@ if [ "$VERBOSE" == true ]; then
     echo "Installed tools: $(ls "$TOOL_DEST")"
 fi
 
-if [ "$DEVCONTAINER" == true ]; then 
+# python packages to be installed for helm validation script
+# using `--break-system-packages` here since the python3 env is externally managed through the OS, so we have to pass this to install the dependency for local environment.
+pip3 install deepdiff --break-system-packages
```

Comment:
@super-harsh pointed out a few challenges with requirements.txt in the Dockerfile.

1. Docker build can't easily access files outside of the build context (`.devcontainer` folder).
2. Docker build doesn't share context with the host easily, which means if we were to create a venv here it probably wouldn't exist on the host filesystem. It would be nice if it could exist on the host filesystem though as that way it works in builds and for IDEs (for those like me using the host for their IDE)

I think the right thing to do here is to split the management of the Python dependencies required into two sections.
**Section 1:** In the Dockerfile/setup.sh, install virtualenv - this you have already and is required to use venv on either host or docker container so it makes sense to be in the main Dockerfile setup.
**Section 2:** Put requirements.txt in `v2/scripts` and use it _in the taskfile_, not in the Dockerfile. I think this makes sense because we could in theory have multiple Python projects with multiple sets of dependencies so doing a single set of Python dependency management in the Dockerfile is wrong. In the same way that we don't install the Go dependencies for our projects in the Dockerfile - we let the Go tooling do that when we run `go build` via the Taskfile.

I think accomplishing this boils down to making a Taskfile target that does something like:
```
  scripts:build-venv:
    desc: Creates a Python virtual env
    dir: {{.SCRIPTS_ROOT}}
    cmds:
      - python3 -m virtualenv venv
      - source venv/bin/activate
      - pip install -r requirements.txt
      - deactivate
```

...
  controller:validate-helm:
    desc: Validates and compares helm against the ASO yaml generated by kustomize and ensures that they match
    dir: "{{.CONTROLLER_ROOT}}"
    deps:
      - task: controller:run-kustomize
        vars:
          VERSION_TAG: "{{.LATEST_VERSION_TAG}}"
          CONTROLLER_IMAGE: "azureserviceoperator:{{.LATEST_VERSION_TAG}}"
      - task: scripts:build-venv
    cmds:
      - defer:
          deactivate  # deactivate the venv even if the script fails
      - source venv/bin/activate # activate the venv
      - "{{.SCRIPTS_ROOT}}/validate_helm.py --helm-dir=$PWD/charts/azure-service-operator --yaml-path=$PWD/bin/azureserviceoperator_{{.LATEST_VERSION_TAG}}.yaml"
```